model/licensing.py

```diff
@@ -935,16 +934,11 @@ def calculate_work(
             work.license_pools.append(self)
             licensepools_changed = True
 
-        # Recalculate the display information for the Work, since the
+        # Recalculate the display information for the Work. Either the
         # associated LicensePools have changed, which may have caused
-        # the Work's presentation Edition to change.
-        #
-        # TODO: In theory we can speed things up by only calling
-        # calculate_presentation if licensepools_changed is
-        # True. However, some bits of other code call calculate_work()
-        # under the assumption that it always calls
-        # calculate_presentation(), so we'd need to evaluate those
-        # call points first.
```

Comment:
We're now heavily relying on the fact that LicensePool.calculate_work() always calls Work.calculate_presentation(), but the tradeoff is that we should now be calling LicensePool.calculate_work() a lot less often.